app/src/main/java/org/dicekeys/app/openpgp/PublicPacket.kt

```diff
@@ -0,0 +1,55 @@
+package org.dicekeys.app.openpgp;
+
+import com.google.common.io.ByteStreams
+import org.bouncycastle.crypto.params.Ed25519PrivateKeyParameters
+import org.dicekeys.app.extensions.toHex
+import java.math.BigInteger
+import java.nio.ByteBuffer
+import java.security.MessageDigest
+
+class PublicPacket(private val publicKey: ByteArray, private val timestamp: UInt): Packet() {
+
+    override val ctb: Int
+        get() = 0x98
+
+    override val body: ByteArray by lazy {
+        val body = ByteStreams.newDataOutput()
+
+        val taggedPublicKey = byteArrayOf(0x40) + publicKey
+
+        body.write(Version)
+        body.writeInt(timestamp.toInt())
+        body.write(Ed25519Algorithm)
+        body.write(Ed25519CurveOid.size)
+        body.write(Ed25519CurveOid)
+
+        body.write(Mpi.fromByteArray(taggedPublicKey).toByteArray())
+
+        body.toByteArray()
+    }
+
+    fun fingerprint(): ByteArray {
+        val out = ByteStreams.newDataOutput()
+        val digest: MessageDigest = MessageDigest.getInstance("SHA-1")
```

Comment:
Is there an option for a non-SHA-1 fingerprint?  SHA-1 is considered broken by the crypto community.

Reply:
No, there is not option to change the fingerprint hash algo.
From RFC2440 (11.2)

> Note that it is possible for there to be collisions of key IDs -- two
   different keys with the same key ID. Note that there is a much
   smaller, but still non-zero probability that two different keys have
   the same fingerprint.